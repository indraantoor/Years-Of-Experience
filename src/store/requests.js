import { collection, getDocs, doc, updateDoc, getDoc } from "@firebase/firestore";
import { db } from "../firebase-config";

export const fetchExperiencesRequest = async (userId) => {
  const workExperiencesRef = collection(db, "users", userId, "workExperiences");
  const workExperiences = await getDocs(workExperiencesRef);
  const response = workExperiences.docs.map((doc) => ({
    ...doc.data(),
    id: doc.id,
  }));
  return response;
};

export const fetchUserDetailsRequest = async (userId) => {
  const userRef = doc(db, "users", userId);
  const user = await getDoc(userRef);
  const response = { ...user.data(), id: userId };
  return response;
};