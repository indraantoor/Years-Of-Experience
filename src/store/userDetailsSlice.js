import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
import { doc, getDoc, updateDoc } from "@firebase/firestore";
import { db } from "../firebase-config";
import {
  fetchUserDetailsFromApi,
  updateUserDetailsToApi,
} from "./helpers/userDetailsSliceHelpers";

const initialState = {
  loading: true,
  error: false,
  redirect: false,
};

export const userDetailsSlice = createSlice({
  name: "userDetails",
  initialState,
  reducers: {
    update: updateDetails,
  },
  extraReducers: {
    [fetchUserDetailsFromApi.pending]: (state, action) => {
      state.loading = true;
      state.error = false;
      state.redirect = false;
    },
    [fetchUserDetailsFromApi.fulfilled]: (state, action) => {
      updateDetails(state, action);
      state.loading = false;
      state.error = false;
      state.redirect = false;
    },
    [fetchUserDetailsFromApi.rejected]: (state, action) => {
      state.loading = false;
      state.error = true;
      state.redirect = false;
    },
    [updateUserDetailsToApi.pending]: (state, action) => {
      state.loading = true;
      state.error = false;
      state.redirect = false;
    },
    [updateUserDetailsToApi.fulfilled]: (state, action) => {
      state.loading = false;
      state.error = false;
      state.redirect = true;
    },
    [updateUserDetailsToApi.rejected]: (state, action) => {
      state.loading = false;
      state.error = true;
      state.redirect = false;
    },
  },
});

function updateDetails(state, action) {
  for (let property in action.payload) {
    state[property] = action.payload[property];
  }
}
