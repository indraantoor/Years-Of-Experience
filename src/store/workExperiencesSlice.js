import { createSlice } from "@reduxjs/toolkit";
import {
  fetchWorkExperiencesFromApi,
  updateWorkExperienceToApi,
  updateWorkExperience,
  deleteWorkExperience,
} from "./helpers/workExperiencesSliceHelpers";

const initialState = {
  data: [],
  loading: true,
  error: false,
  redirect: false,
};

const reducers = {
  update: updateWorkExperience,
  delete: deleteWorkExperience,
};

export const workExperiencesSlice = createSlice({
  name: "workExperiences",
  initialState,
  reducers: reducers,
  extraReducers: {
    [fetchWorkExperiencesFromApi.pending]: (state, action) => {
      state.loading = true;
      state.error = false;
      state.redirect = false;
    },
    [fetchWorkExperiencesFromApi.fulfilled]: (state, action) => {
      state.data = action.payload;
      state.loading = false;
      state.error = false;
      state.redirect = false;
    },
    [fetchWorkExperiencesFromApi.rejected]: (state, action) => {
      state.loading = false;
      state.error = true;
      state.redirect = false;
    },
    [updateWorkExperienceToApi.pending]: (state, action) => {
      state.loading = true;
      state.error = false;
      state.redirect = false;
    },
    [updateWorkExperienceToApi.fulfilled]: (state, action) => {
      state.loading = false;
      state.error = false;
      state.redirect = true;
    },
    [updateWorkExperienceToApi.rejected]: (state, action) => {
      state.loading = false;
      state.error = true;
      state.redirect = false;
    },
  },
});

export default workExperiencesSlice.reducer;
