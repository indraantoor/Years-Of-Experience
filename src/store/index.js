import { configureStore } from "@reduxjs/toolkit";
import { userDetailsSlice } from "./userDetailsSlice";
import { workExperiencesSlice } from "./workExperiencesSlice";

export const store = configureStore({
  reducer: {
    userDetails: userDetailsSlice.reducer,
    workExperiences: workExperiencesSlice.reducer,
  },
});
