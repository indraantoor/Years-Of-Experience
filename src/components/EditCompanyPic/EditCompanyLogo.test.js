/**
 *  @jest-environment jsdom
 */
import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import { EditCompanyPic } from ".";
import "@testing-library/jest-dom";
import { StaticRouter } from "react-router-dom/server";
import { create } from "react-test-renderer";

import { Provider } from "react-redux";
import { store } from "../../store/index";

jest.mock("@firebase/firestore", () => ({
  doc: "hell",
  getDoc: "jk",
  updatedoc: "jkjkj",
  initializeApp: (m) => {},
  getFirestore: "dkendk",
  enableIndexedDbPersistence: "nknkn",
}));

jest.mock("@firebase/storage", () => ({
  ref: "",
  uploadBytesResumable: "",
  getDownloadURL: "",
}));

jest.mock("../../firebase-config.js", () => ({
  getStorage: "nknk",
}));

test("renders correctly", () => {
  render(
    <StaticRouter>
      <Provider store={store}>
        <EditCompanyPic />
      </Provider>
    </StaticRouter>
  );

  const tree = create(
    <StaticRouter>
      <Provider store={store}>
        <EditCompanyPic />
      </Provider>
    </StaticRouter>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
