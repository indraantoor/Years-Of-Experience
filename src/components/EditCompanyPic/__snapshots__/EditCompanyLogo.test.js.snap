// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bczRLJ hXvkvx"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="image-container"
    >
      <img
        alt="profile avatar"
      />
    </div>
    <label
      htmlFor="profileimg"
    >
      Company Logo
      <input
        id="profileimg"
        onChange={[Function]}
        type="file"
      />
    </label>
    <button
      type="submit"
    >
      Update
    </button>
  </form>
</div>
`;
